app/features/theme/color-mode-toggle.tsx

```diff
@@ -0,0 +1,38 @@
+import { Button } from '~/components/ui/button';
+import { colorModes } from './theme.constants';
+import type { ColorMode } from './theme.types';
+import { useTheme } from './use-theme';
+
+export function ColorModeToggle() {
+  const { colorMode, setColorMode } = useTheme();
+
+  const handleToggle = () => {
+    // Cycle through modes: light -> dark -> system
+    const currentIndex = colorModes.indexOf(colorMode);
+    const nextIndex = (currentIndex + 1) % colorModes.length;
+    setColorMode(colorModes[nextIndex]);
+  };
+
+  const getIcon = (mode: ColorMode) => {
+    switch (mode) {
+      case 'light':
+        return '☀️';
+      case 'dark':
+        return '🌙';
+      case 'system':
+        return '💻';
+      default:
+        return '☀️';
+    }
+  };
+
+  return (
+    <Button
```

Comment:
it seems very unusual that you have a button for this and then you are click in to switch between 3 different options. something like what is implemented [here](https://shadcn-ui-themes.vercel.app/music) makes more sense imo

Reply:
made it a  bit fancier in #216